temporian/implementation/numpy/operators/assign.py

```diff
@@ -58,11 +63,11 @@ def __call__(
                 " index."
             )
 
-        output = NumpyEvent(
-            data=left_event.data.copy(), sampling=left_event.sampling
-        )
+        output = NumpyEvent(data={}, sampling=left_event.sampling)
 
-        for index in left_event.data.keys():
+        for index, left_features in left_event.data.items():
+            # Copy the features of left to the output event
+            output.data[index] = [feature for feature in left_features]
             output_data = output.data[index]
```

Comment:
this assignment was a bit hard to understand - what do you think about defining `data = {}` before the for loop without creating the `output` event, and then creating the event when `data` is already created after the for loop?

Reply:
sounds pretty